pyproject.toml

```diff
@@ -1,18 +1,59 @@
 [build-system]
-requires = [
-    "scikit-build>=0.12",
-    "setuptools>=42",
+requires = ["scikit-build-core"]
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "cmake"
+version = "0.0.1"
+description = "CMake is an open-source, cross-platform family of tools designed to build, test and package software"
+keywords = ["CMake", "build", "c++", "fortran", "cross-platform", "cross-compilation"]
+readme = "README.rst"
+license = {text = "Apache 2.0"}
+authors = [
+    {name = "Jean-Christophe Fillion-Robin", email = "jchris.fillionr@kitware.com"},
+]
+classifiers = [
+    "License :: OSI Approved :: Apache Software License",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: C",
+    "Programming Language :: C++",
+    "Programming Language :: Fortran",
+    "Programming Language :: Python",
+    "Operating System :: OS Independent",
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development :: Build Tools",
+    "Typing :: Typed"
+]
+requires-python = ">=3.7"
```

Comment:
This might be a showstopper (for now) for this project ?

Reply:
https://github.com/scikit-build/scikit-build-core/issues/39 is needed. It won't ever build from SDist on Python <3.7, but don't think that's a big deal at this point.

Reply:
> https://github.com/scikit-build/scikit-build-core/issues/39 is needed

It would be nice to have it but it's not needed. It would allow to get rid of all the specific repair scripts in the repo.

> It won't ever build from SDist on Python <3.7, but don't think that's a big deal at this point.

I'm a bit more hesitant on "runs everywhere", "need >= 3.7 to build from sdist". In the end, it will be ok but might be a bit too soon ?

Reply:
If we build from SDist, I think it's already much slower, more likely to fail, etc - it's really supposed to be a wheel distribution. So not supporting building from SDist on <=3.6 doesn't seem too bad. And a user can always either use Python 3.7+ to build a wheel then install it on 2.7-3.6, or use an older version.